contracts/core/SetToken.sol

```diff
@@ -211,13 +211,17 @@ contract SetToken is
     }
 
     function burn(
-        uint quantity
+        address _from,
+        uint _quantity
     )
         external
+        isCore
+        isNonZero(_quantity)
     {
-        balances[msg.sender] = balances[msg.sender].sub(quantity);
-        totalSupply_ = totalSupply_.sub(quantity);
-        emit Transfer(msg.sender, address(0), quantity);
+        require(balances[_from] >= _quantity);
+
+        balances[_from] = balances[_from].sub(_quantity);
+        totalSupply_ = totalSupply_.sub(_quantity);
```

Comment:
We should add an event for Transfer from the user to a nonexistent address.
```
emit Transfer(from, address(0), amount);
```

```

Reply:
added